src/Workspaces/Core/Portable/Editing/ImportAdderService.cs

```diff
@@ -53,7 +53,7 @@ internal abstract class ImportAdderService : ILanguageService
             var existingNamespaces = new HashSet<INamespaceSymbol>();
             await this.GetExistingImportedNamespacesAsync(document, model, existingNamespaces, cancellationToken).ConfigureAwait(false);
 
-            var namespacesToAdd = new HashSet<INamespaceSymbol>(namespaces);
+            var namespacesToAdd = new HashSet<INamespaceSymbol>(namespaces, NamespaceEqualityComparer.Singleton);
```

Comment:
I did some more testing. Usually, namespaces are of type `Microsoft.CodeAnalysis.CSharp.Symbols.SourceNamespaceSymbol` or `Microsoft.CodeAnalysis.CSharp.Symbols.Metadata.PE.PENestedNamespaceSymbol`. They both return true for reference equality, so the missing override for Equals does not hurt.

The scenario for this issue concerns `Microsoft.CodeAnalysis.CSharp.Symbols.MergedNamespaceSymbol` types (because the target symbol is defined in two assemblies). In that case, reference equality returns false. And because Equals is not overriden, it fails.